src/coreclr/vm/clsload.cpp

```diff
@@ -769,19 +769,10 @@ TypeHandle ClassLoader::LoadConstructedTypeThrowing(TypeKey *pKey,
 
     // Lookup in the classes that this class loader knows about
 
-    if (pKey->HasInstantiation() && ClassLoader::IsTypicalSharedInstantiation(pKey->GetInstantiation()))
-    {
-        _ASSERTE(pKey->GetModule() == ComputeLoaderModule(pKey));
-        typeHnd = pKey->GetModule()->LookupFullyCanonicalInstantiation(pKey->GetTypeToken(), &existingLoadLevel);
-    }
-
-    if (typeHnd.IsNull())
+    typeHnd = LookupTypeHandleForTypeKey(pKey);
+    if (!typeHnd.IsNull())
     {
-        typeHnd = LookupTypeHandleForTypeKey(pKey);
-        if (!typeHnd.IsNull())
-        {
-            existingLoadLevel = typeHnd.GetLoadLevel();
-        }
+        existingLoadLevel = typeHnd.GetLoadLevel();
```

Comment:
I think the if block below and the declaration of `existingLoadLevel` can be moved into this now.

Reply:
We don't even need a declaration for `existingLoadLevel`.